### ISSUE TITLE
[Bug]: Staging Only - When Grant is assigned to a team, all teams assigned receive notification

### ISSUE THREAD

**Issue by ClaireValdivia** [usdigitalresponse/usdr-gost#3107]
### Why is this issue important?

Assignment notifications are not sending correctly and will be confusing to users

### Current State

**Steps to recreate:**
Two users from two different teams, both have notifications enabled - one belongs to team USDR and one belongs to team Claire Test.
- Assign grant to team USDR - the user in this team recieves a notification of the assignment
- Assign the same grant to team Claire Test - both users from both teams will receive assignment email
- For every new assignment made, every person who has be previously assigned the grant will receive another notification

### Expected State

A user will only receive a notification when their team is assigned, not when another team is assigned

### Implementation Plan

The following functions need to be added...

**Comment by jeffsmohan** [comment]
@as1729 Personally, I found the API design here unexpected, which threw me off the case a bit this morning. Specifically, I would normally expect to find an endpoint like `/:grantId/assign/agencies` to behave something like this: 

- `GET`: returns the list of assigned agencies for the grant (current behavior)
- `POST`: expects a single `agency_id` to add to the list of assigned agencies (NEW endpoint)
- `PUT`: expects an `agency_ids` array and resets the list of assigned agencies to the supplied list (existing endpoint, but CHANGED behavior — the current endpoint effectively does a set union between the existing assignments and the provided `agency_ids`, which feels like unexpected behavior)
- `DELETE`: expects a single `agency_id` to remove from the list of assigned agencies (current behavior)
- [note that there's no PUT action, which would be a change]

And fwiw, I don't know of any good current use cases for the PUT behavior above, so I would fully remove it. Instead, when a user assigns a grant, the frontend should hit the POST endpoint described above with just the single added agency ID. Then we can easily have the server trigger emails just for that given agency ID as part of the route handler for that endpoint. 

So to summarize, I would propose a third implementation path: 

1. Add a `POST /:grantId/assign/agencies` endpoint that expects a single agency ID, assigns it, and triggers emails just for that agency
2. Update the frontend to hit this endpoint instead of the existing PUT endpoint
3. Remove the PUT endpoint

What do you think?

**Comment by as1729** [comment]
This is definitely a cleaner implementation rather than building on an already confusing existing api. Let's go with your approach. Thank you!

**Comment by ClaireValdivia** [comment]
This is working as expected in staging! Assigned a grant to two teams, and users only received the notification when assigned to their team.
